core/src/engine/coro/pool.hpp

```diff
@@ -119,31 +136,53 @@ typename Pool<Task>::CoroutinePtr Pool<Task>::GetCoroutine() {
       return *this;
     }
   };
-
   std::optional<Coroutine> coroutine;
   CoroutineMover mover{coroutine};
 
   // First try to dequeue from 'working set': if we can get a coroutine
   // from there we are happy, because we saved on minor-page-faulting (thus
   // increasing resident memory usage) a not-yet-de-virtualized coroutine stack.
-  if (used_coroutines_.try_dequeue(
-          GetUsedPoolToken<moodycamel::ConsumerToken>(), mover) ||
-      initial_coroutines_.try_dequeue(mover)) {
-    --idle_coroutines_num_;
+  if (!local_coro_buffer_.empty() || TryPopulateLocalCache()) {
+    coroutine = std::move(local_coro_buffer_.back());
+    local_coro_buffer_.pop_back();
+  } else if (initial_coroutines_.try_dequeue(mover)) {
+    idle_coroutines_num_.fetch_sub(1, std::memory_order_release);
```

Comment:
On x86*, `std::memory_order_release` here will not matter, but will raise questions. Let's remove it

Reply:
leave memory order in case idle_coroutines_num_.load(std::memory_order_acquire) ?